core/src/main/java/com/github/phantazmnetwork/core/inventory/InventoryAccess.java

```diff
@@ -0,0 +1,19 @@
+package com.github.phantazmnetwork.core.inventory;
+
+import net.kyori.adventure.key.Key;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Unmodifiable;
+
+import java.util.Map;
+import java.util.Objects;
+
+public record InventoryAccess(@NotNull InventoryProfile profile,
```

Comment:
might be worth it to make this an interface, would allow different implementations if desirable

Reply:
not sure why it would ever be one really?